### PR TITLE
add microsoft vuln policy for CVE-2023-21716

### PR DESCRIPTION
**Pull request by atomic111**

### CODE REVIEW

**Review by chris-rock — APPROVED** [review]
Great work @atomic111

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            packages.where( name == /SharePoint/ ).length > 0
+        scoring_queries:
+          mondoo-microsoft-vulnerability-sharepoint-CVE-2023-21716:
+queries:
+  - uid: mondoo-microsoft-vulnerability-office-CVE-2023-21716
+    title: Ensure Microsoft Word Remote Code Execution Vulnerability for Office is not on the system
+    severity: 98
+    docs:
+      desc: |
+        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruptionand allows attackers to achieve remote code execution with the privileges of the victim that opens a malicious RTF document. Mor information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```
Comment (tas50) [inline core/mondoo-microsoft-vulnerability.mql.yaml:55]:
```suggestion
        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      remediation: |
+        Please update your Office installation or use the workaround. More information can be found (https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21716)[https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21716].
```
Comment (tas50) [inline core/mondoo-microsoft-vulnerability.mql.yaml:57]:
```suggestion
        Please update your Office installation or use the workaround. More information can be found at (https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21716)[https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21716].
```

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      packages.where( name == /Office/ && name == /2016|2019|2021/ ).all(version.split('.')[0] == 16 && version.split('.')[2] >= 16026)
+    refs:
+      - title: Release notes for Microsoft Office security updates
+        url: https://learn.microsoft.com/en-us/officeupdates/microsoft365-apps-security-updates
+  - uid: mondoo-microsoft-vulnerability-office-CVE-2023-21716-workaround
+    title: Block opening RTF documents (workaround)
+    severity: 98
+    docs:
+      desc: |
+        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruptionand allows attackers to achieve remote code execution with the privileges of the victim that opens a malicious RTF document. Mor information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```
Comment (tas50) [inline core/mondoo-microsoft-vulnerability.mql.yaml:68]:
```suggestion
        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```

**Comment by atomic111** [comment]
@tas50 thanks for the fixes

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            packages.where( name == /SharePoint/ ).length > 0
+        scoring_queries:
+          mondoo-microsoft-vulnerability-sharepoint-CVE-2023-21716:
+queries:
+  - uid: mondoo-microsoft-vulnerability-office-CVE-2023-21716
+    title: Ensure Microsoft Word Remote Code Execution Vulnerability for Office is not on the system
+    severity: 98
+    docs:
+      desc: |
+        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```
Comment (tas50) [inline core/mondoo-microsoft-vulnerability.mql.yaml:55]:
```suggestion
        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found at (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      packages.where( name == /Office/ && name == /2016|2019|2021/ ).all(version.split('.')[0] == 16 && version.split('.')[2] >= 16026)
+    refs:
+      - title: Release notes for Microsoft Office security updates
+        url: https://learn.microsoft.com/en-us/officeupdates/microsoft365-apps-security-updates
+  - uid: mondoo-microsoft-vulnerability-office-CVE-2023-21716-workaround
+    title: Block opening RTF documents (workaround)
+    severity: 98
+    docs:
+      desc: |
+        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```
Comment (tas50) [inline core/mondoo-microsoft-vulnerability.mql.yaml:68]:
```suggestion
        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found at (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      registrykey.property(path: 'HKEY_CURRENT_USER\Software\Microsoft\Office\15.0\Word\Security\FileBlock', name: 'OpenInProtectedView').value == 0 || registrykey.property(path: 'HKEY_CURRENT_USER\Software\Microsoft\Office\16.0\Word\Security\FileBlock', name: 'OpenInProtectedView').value == 0
+    refs:
+      - title: Microsoft Word Remote Code Execution Vulnerability
+        url: https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-21716
+  - uid: mondoo-microsoft-vulnerability-sharepoint-CVE-2023-21716
+    title: Ensure Microsoft Word Remote Code Execution Vulnerability for SharePoint is not on the system
+    severity: 98
+    docs:
+      desc: |
+        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```
Comment (tas50) [inline core/mondoo-microsoft-vulnerability.mql.yaml:108]:
```suggestion
        The critical vulnerability CVE-2023-21716 is a Microsoft Word RTF Font Table Heap Corruption and allows attackers to achieve remote code execution with the victim's privileges that opens a malicious RTF document. More information can be found at (https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md)[https://qoop.org/publications/cve-2023-21716-rtf-fonttbl.md]
```

File: core/mondoo-microsoft-vulnerability.mql.yaml [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+      mondoo.com/category: security
+    docs:
+      desc: |
+        ## Overview
+
+        Mondoo Microsoft Vulnerability Policy checks for Windows and Microsoft Application vulnerabilities. It should be used in combination with the Platform Vulnerability Policy to identify missing patches.
+
+        ### Run policy
+
+        To run this policy against VMware vCenter:
```
[the remark this answers is not in the document]
Comment (atomic111) [reply core/mondoo-microsoft-vulnerability.mql.yaml]: @tas50 thanks for the hint. fixed

**Pull request merged**